workbench/src/menu.c

```diff
@@ -0,0 +1,310 @@
+{
+	workbench_free(wb_globals.opened_wb);
+	wb_globals.opened_wb = NULL;
+
+	menu_item_new_activate();
+	menu_item_open_activate();
+	menu_item_save_deactivate();
+	menu_item_settings_deactivate();
+	menu_item_close_deactivate();
+
+	sidebar_update(SIDEBAR_CONTEXT_WB_CLOSED, NULL);
+}
+
+
+/** Setup the workbench menu.
+ *
+ **/
+gboolean menu_init(void)
+{
+	/* Create menu and root item/label */
+	menu_data.menu = gtk_menu_new();
+	menu_data.root_item = gtk_menu_item_new_with_label(_("Workbench"));
+	gtk_widget_show(menu_data.root_item);
+
+	/* Create new menu item "New Workbench" */
+	menu_data.item_new = gtk_menu_item_new_with_mnemonic(_("New"));
+	gtk_widget_show(menu_data.item_new);
+	gtk_menu_append(GTK_MENU (menu_data.menu), menu_data.item_new);
+	g_signal_connect(menu_data.item_new, "activate",
+					 G_CALLBACK(item_new_workbench_activate_cb), NULL);
+	geany_plugin_set_data(wb_globals.geany_plugin, menu_data.item_new, NULL);
+
+	/* Create new menu item "Open Workbench" */
+	menu_data.item_open = gtk_menu_item_new_with_mnemonic(_("Open"));
+	gtk_widget_show(menu_data.item_open);
+	gtk_menu_append(GTK_MENU (menu_data.menu), menu_data.item_open);
+	g_signal_connect(menu_data.item_open, "activate",
+					 G_CALLBACK(item_open_workbench_activate_cb), NULL);
+	geany_plugin_set_data(wb_globals.geany_plugin, menu_data.item_open, NULL);
+
+	/* Create new menu item "Save Workbench" */
+	menu_data.item_save = gtk_menu_item_new_with_mnemonic(_("Save"));
+	gtk_widget_show(menu_data.item_save);
+	gtk_menu_append(GTK_MENU (menu_data.menu), menu_data.item_save);
+	g_signal_connect(menu_data.item_save, "activate",
+					 G_CALLBACK(item_save_workbench_activate_cb), NULL);
+	geany_plugin_set_data(wb_globals.geany_plugin, menu_data.item_save, NULL);
+
+	/* Create new menu item "Workbench Settings" */
+	menu_data.item_settings = gtk_menu_item_new_with_mnemonic(_("Settings"));
+	gtk_widget_show(menu_data.item_settings);
+	gtk_menu_append(GTK_MENU (menu_data.menu), menu_data.item_settings);
+	g_signal_connect(menu_data.item_settings, "activate",
+					 G_CALLBACK(item_workbench_settings_activate_cb), NULL);
+	geany_plugin_set_data(wb_globals.geany_plugin, menu_data.item_settings, NULL);
+
+	/* Create new menu item "Close Workbench" */
+	menu_data.item_close = gtk_menu_item_new_with_mnemonic(_("Close"));
+	gtk_widget_show(menu_data.item_close);
+	gtk_menu_append(GTK_MENU (menu_data.menu), menu_data.item_close);
+	g_signal_connect(menu_data.item_close, "activate",
+					 G_CALLBACK(item_close_workbench_activate_cb), NULL);
+	geany_plugin_set_data(wb_globals.geany_plugin, menu_data.item_close, NULL);
+
+	/* Add our menu to the main window (left of the help menu) */
+	gtk_menu_item_set_submenu(GTK_MENU_ITEM(menu_data.root_item), menu_data.menu);
+	gtk_menu_shell_insert
+		(GTK_MENU_SHELL (ui_lookup_widget(wb_globals.geany_plugin->geany_data->main_widgets->window, "menubar1")),
+		 menu_data.root_item, 8);
```

Comment:
You could use `gtk_container_get_children()` and use the list's length to figure out how many items there are, and then stick yours at -2 from the end. You could also use `gtk_container_get_children()` and iterate through looking for the item with the name from the Glade file for the Help menu (not guaranteed stable, but unlikely to change), and do how you said. Geany has no really good way to do this.